core/src/main/java/io/snappydata/impl/SnappyHiveCatalog.java

```diff
@@ -263,8 +285,34 @@ public String catalogSchemaName() {
     return SnappyStoreHiveCatalog.HIVE_METASTORE();
   }
 
+  @Override
+  public void clearCache(String schemaName, String tableName) {
+    if (schemaName == null && tableName == null) {
+      this.nonExistentTables.invalidateAll();
+      this.tableCache.invalidateAll();
+      // also clear "isRowBuffer" for all regions
+      for (PartitionedRegion pr : PartitionedRegion.getAllPartitionedRegions()) {
+        pr.clearIsRowBuffer();
+      }
+    } else {
+      // hive is case-insensitive
+      AbstractMap.SimpleEntry<String, String> key = new AbstractMap.SimpleEntry<>(
+          Utils.toUpperCase(schemaName), Utils.toUpperCase(tableName));
+      this.nonExistentTables.invalidate(key);
+      this.tableCache.invalidate(key);
+      // also clear "isRowBuffer" since it may have been cached incorrectly
+      // when column store was still being created
+      Region<?, ?> region = Misc.getRegion(Misc.getRegionPath(key.getKey(),
+          key.getValue(), null), false, true);
+      if (region instanceof PartitionedRegion) {
+        ((PartitionedRegion)region).clearIsRowBuffer();
```

Comment:
When is the isRowBuffer set BTW?

Reply:
Its initialised lazily on first access. See store PR: https://github.com/SnappyDataInc/snappy-store/pull/444/files#diff-0bd50a396f29b0cd9abd2865f498a5b2L2526

That was a bug because it could access before child column store regions are created then it will be permanently wrong.

Reply:
I saw the problem in manual testing while manually querying systables while tables were being created. Then the tableType would go permanently wrong for few tables, but the same could happen due to internal calls too and user will probably not notice (except that rollover, snapshot would no longer happen).